### ISSUE TITLE
Question about states and data

### ISSUE THREAD

**Issue by kielby** [boost-ext/sml#566]
Is the statement on the [tutorial page](https://boost-ext.github.io/sml/tutorial.html)outdated?

> SML states cannot have data as data is injected directly into guards/actions instead.

On the examples page in the data section it shows otherwise right? Or is the statement above regarding something else?
https://boost-ext.github.io/sml/examples.html#data

**Comment by MatthiasKleesSchulz** [comment]
Nope, the tutorial is not outdated. It is true, the STATE itself cannot have data. The example shows how u can store data in the STATEMACHINE. An other solution is to work with dependencies.

**Comment by martinerk0** [comment]
@MatthiasKleesSchulz Can I ask what's the difference between "states cannot have data" and "Another solution is to work with dependencies"?
